pdns/recursordist/rec-taskqueue.cc

```diff
@@ -22,50 +22,139 @@
 #include "rec-taskqueue.hh"
 #include "taskqueue.hh"
 #include "lock.hh"
-#include "logger.hh"
+#include "logging.hh"
 #include "stat_t.hh"
 #include "syncres.hh"
 
-static LockGuarded<pdns::TaskQueue> s_taskQueue;
-static pdns::stat_t s_almost_expired_tasks_pushed;
-static pdns::stat_t s_almost_expired_tasks_run;
-static pdns::stat_t s_almost_expired_tasks_exceptions;
+// For rate lmiting, we maintain a set of tasks recently submitted.
+class TimedSet
+{
+public:
+  TimedSet(time_t t) :
+    d_expiry_seconds(t)
+  {
+  }
+
+  uint64_t purge(time_t now)
+  {
+    // This purge is relatively sleep, as we're walking an ordered index
```

Comment:
sleep?

Reply:
fast, I guess, as indeed we `break` the moment we hit things that are still new enough

Reply:
I meant to type cheap, but my fingers are autonomous beings